doc/platform_api/CMIS_Diagnostic_Monitoring_Overview_in_SONiC.md

```diff
@@ -0,0 +1,971 @@
+3. With every timer expiry, the `DomInfoUpdateTask` thread reads the diagnostic information from the transceiver and updates the relevant tables in `redis-db` for all the ports. The steps performed to update the diagnostic information for a port are as follows:
+    1. Ensure DOM monitoring is enabled for the port. If DOM monitoring is disabled, skip updating the diagnostic information for the port.
+    2. If the current port is the first port from its breakout port group to be polled, clear the cached diagnostic information. For all subsequent ports in the breakout port group, use the cached diagnostic information to update the `redis-db`.
+    3. Read the transceiver firmware information from the module and update the `TRANSCEIVER_FIRMWARE_INFO` table.
+    4. Read the transceiver DOM sensor data from the module and update the `TRANSCEIVER_DOM_SENSOR` table.
+    5. Read the transceiver DOM flag data from the module, record the timestamp, and update the `TRANSCEIVER_DOM_FLAG` table.
+    6. Analyze the transceiver DOM flag data by comparing the current flag data with the previous flag data and update the `TRANSCEIVER_DOM_FLAG_CHANGE_COUNT`, `TRANSCEIVER_DOM_FLAG_TIME_SET`, and `TRANSCEIVER_DOM_FLAG_TIME_CLEAR` tables.
+    7. Read the transceiver status data from the module and update the `TRANSCEIVER_STATUS` table.
+    8. If the transceiver supports VDM monitoring, perform the following steps:
+        1. Freeze the statistics by calling the CMIS API and wait for `FreezeDone`. Once the statistics are frozen, record the timestamp and copy the VDM and PM statistics from the transceiver.
+        2. Unfreeze the statistics by calling the CMIS API.
+        3. Update the `TRANSCEIVER_VDM_SAMPLE` and `TRANSCEIVER_PM` tables with the VDM and PM statistics respectively.
+        4. Analyze the VDM flags by comparing the current statistics with the previous statistics and update the `TRANSCEIVER_VDM_FLAG`, `TRANSCEIVER_VDM_FLAG_CHANGE_COUNT`, `TRANSCEIVER_VDM_FLAG_TIME_SET`, and `TRANSCEIVER_VDM_FLAG_TIME_CLEAR` tables.
+        5. Analyze the PM flags by comparing the current statistics with the previous statistics and update the `TRANSCEIVER_PM_FLAG`, `TRANSCEIVER_PM_FLAG_CHANGE_COUNT`, `TRANSCEIVER_PM_FLAG_TIME_SET`, and `TRANSCEIVER_PM_FLAG_TIME_CLEAR` tables.
+
+#### 4.2.2 Details of Flag Analysis of Tables
+
+**Note**: For simplicity, this section uses VDM as an example. However, the same analysis is applicable for DOM and PM as well.
+
+**Purpose of Flag Analysis:**
+
+The purpose of flag analysis is to track the status of various parameters and to count the number of times each VDM flag has changed. It also records the timestamp when each VDM flag was set and cleared.
+
+**Tables Used for Flag Analysis:**
+
+- `TRANSCEIVER_VDM_FLAG`: This table stores flags indicating the status of various VDM parameters.
+- `TRANSCEIVER_VDM_FLAG_CHANGE_COUNT`: This table keeps a count of how many times each VDM flag has changed.
```

Comment:
If xcvrd stopped, will the change count be reset to 0? The same question to time set and time clear. Also, should we record the change count to syslog?